lib/modules/manager/mix/extract.ts

```diff
@@ -64,17 +67,39 @@ export async function extractPackageFile(
           }
         }
 
-        deps.push(dep);
+        deps.set(app, dep);
+        logger.trace({ dep }, `setting ${app}`);
         depMatchGroups = depMatchRegExp.exec(depBuffer)?.groups;
       }
     }
   }
-  const res: PackageFileContent = { deps };
   const lockFileName =
     (await findLocalSiblingOrParent(packageFile, 'mix.lock')) ?? 'mix.lock';
-  // istanbul ignore if
-  if (await localPathExists(lockFileName)) {
-    res.lockFiles = [lockFileName];
+  const lockFileContent = await readLocalFile(lockFileName, 'utf8');
+
+  if (lockFileContent) {
+    const lockFileLines = lockFileContent.split(newlineRegex).slice(1, -1);
+
+    const lockedVersions: Map<string, string> = new Map();
+
+    for (const line of lockFileLines) {
+      const groups = lockedVersionRegExp.exec(line)?.groups;
+      if (groups?.app && groups?.lockedVersion) {
+        lockedVersions.set(groups.app, groups.lockedVersion);
+      }
+    }
+
+    for (const [app, dep] of deps.entries()) {
+      dep.lockedVersion = lockedVersions.get(app);
+      logger.trace(`Found ${dep.lockedVersion} for ${app}`);
```

Comment:
wrong, check if the locked version has this entry.

Reply:
Wouldn't have the same end result when checking for inclusion in `lockedVersions`?

`lockedVersions.get(app)` will return a string or undefined, both of which are valid types of `PackageDependency.lockedVersion`. The only difference is that the key will exist or not exist in the `dep` object.

The trace would either show that no locked version was found, "Found undefined for some_app", or the locked version, "Found 3.2.1 for some_app".

If you would like me to check, could you clarify where the inclusion check should go? Around both lines or just the trace?

Reply:
the message seems to be a little misleading/ confusing then 🤔

Reply:
Updated to only print if a version was found.